### ISSUE TITLE
FEATURE REQUEST: Implement audio cues for when something finishes/fails running on the integrated terminal

### ISSUE THREAD

**Issue by saikatdas0790** [microsoft/vscode#170622]
Right now, the integrated terminal has implemented the shell integration that allows it to show the end result of a long running shell command. Including successfully exiting or failure exits.

It would be helpful to have an audio cue specific to those events so that, for example, I can start running my long-running test suite, and when it finishes executing, I get an audio cue while the VS Code window is in the background or is not focused so that we can be notified that it has completed executing.

Additionally, this should have options for setting it so that:
- the audio cue only executes when VS Code is not in focus. So that I'm not constantly being dinged for every command that I execute
- some sort of delay value so that let's say if the delay is set to 10 seconds, only long running tasks that exceed a run time of 10 seconds should sound the audio cue

Thoughts?

**Comment by Triangle4** [comment]
Very much needed.
When building something large like blender or downloading a massive git, would be great to have an audio cue to know when it is done.

**Comment by meganrogge** [comment]
> some sort of delay value so that let's say if the delay is set to 10 seconds, only long running tasks that exceed a run time of 10 seconds should sound the audio cue

Unfortunately, we cannot do this because sounds must be initiated by user gesture (and played within 4 seconds of that).

The fix for that was here, but we had to revert it because in almost all cases, we don't want media to autoplay. https://github.com/microsoft/vscode/pull/179349/files#diff-27653c212e1cfe533e4eb2f7d0d3f89604c9de48a09583b4cbbbcbd08a07da79L254

**Comment by meganrogge** [comment]
> the audio cue only executes when VS Code is not in focus. So that I'm not constantly being dinged for every command that I execute

when VS Code is not in focus, there's not any user gesture and thus the sounds won't play for the same reasons as above

I'll keep this open in case we ever have a way to specify this rule in a custom way. Even then, we might not do it because this would only work on desktop (not web) from what I understand.

**Comment by VSCodeTriageBot** [comment]
This feature request has not yet received the 20 community [upvotes](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) it takes to make to our backlog. 10 days to go. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
:slightly_frowning_face: In the last 60 days, this feature request has received less than 20 community upvotes and we closed it. Still a big Thank You to you for taking the time to create this issue! To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by N-Demir** [comment]
Would be awesome still!
